service/mount.go

```diff
@@ -495,22 +496,22 @@ func unpublishVolume(
 			err.Error())
 	}
 
-	tgtMnt := false
+	var tgtMnt []string
 	privMnt := false
 	for _, m := range mnts {
 		// Added check for sysDevice.FullPath as that is used by multipath mapper
 		if m.Source == sysDevice.RealDev || m.Device == sysDevice.RealDev || m.Device == sysDevice.FullPath {
 			if m.Path == privTgt {
 				privMnt = true
-			} else if m.Path == target {
-				tgtMnt = true
+			} else if m.Path == target || m.Path == dupTarget {
+				tgtMnt = append(tgtMnt, m.Path)
 			}
 		}
 	}
 
-	if tgtMnt {
-		log.WithFields(f).Debug(fmt.Sprintf("Unmounting %s", target))
-		if err := gofsutil.Unmount(ctx, target); err != nil {
+	for _, t := range tgtMnt {
+		log.WithFields(f).Debugf("Unmounting %s", t)
+		if err := gofsutil.Unmount(ctx, t); err != nil {
 			return lastUnmounted, status.Errorf(codes.Internal,
 				"Error unmounting target: %s", err.Error())
 		}
```

Comment:
Is this the correct behavior when an error is encountered.  If there are other mounts that need to be unmounted then what will the state of the node be?

Reply:
In this change we're looking for `/noderoot` prefixed mounts, not errors, so only these additional mounts are unmounted. So far on driver restart we've seen dupMounts with "`/noderoot`" prefix as on line 473.